Modules/_zoneinfo.c

```
@@ -2572,6 +2572,9 @@ update_strong_cache(zoneinfo_state *state, const PyTypeObject *const type,
     }
 
     StrongCacheNode *new_node = strong_cache_node_new(key, zone);
```

Perhaps there should be a comment in `strong_cache_node_new` that it doesn't need to set `MemoryError` on failure of `PyMem_Malloc`, because the cache intentionally does not propagate exceptions, it just fails silently.

---

Done.